dataflux_pytorch/dataflux_iterable_dataset.py

```diff
@@ -111,53 +114,66 @@ def __init__(
                 during initialization.
         """
         super().__init__()
-        self.storage_client = storage_client
-        if not storage_client:
-            self.storage_client = storage.Client(project=project_name, )
-        dataflux_core.user_agent.add_dataflux_user_agent(self.storage_client)
+        self.storage_client = None
+        if OS == LINUX:
```

Comment:
It looks like this means we're silently ignoring the passed-in storage client when the user is on other platforms? If so, should we either raise an exception or log a warning when it's set on other platforms?

In fact, it seems like in some cases it may work for the user and for you on Mac, but it's not entirely clear when, is that right? If so we may want to log a warning saying that setting the storage client is not supported on non-Linux platforms but try to do it anyway

Reply:
Definitely log a warning at very least to let them know it's being ignored. I would even say throwing an error is probably appropriate so they just remove the setting.

Reply:
Mac is unstable with fork (see https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods) so safest option is to move forward with linux only.